CRM/Contact/BAO/Query.php

```diff
@@ -1014,9 +1014,9 @@ public function addHierarchicalElements() {
 
     foreach ($this->_returnProperties['location'] as $name => $elements) {
       $lCond = self::getPrimaryCondition($name);
+      $locationTypeId = is_numeric($name) ? NULL : array_search($name, $locationTypes);
```

Comment:
This just moves the line & makes it clearer how it is calculated. I didn't change usage of this var elsewhere in this PR n the end